overlord/devicestate/handlers.go

```diff
@@ -57,22 +58,58 @@ func useStaging() bool {
 	return osutil.GetenvBool("SNAPPY_USE_STAGING_STORE")
 }
 
-func deviceAPIBaseURL() string {
+func baseURL() string {
 	if useStaging() {
-		return "https://api.staging.snapcraft.io/api/v1/snaps/auth/"
+		return "https://api.staging.snapcraft.io/"
 	}
-	return "https://api.snapcraft.io/api/v1/snaps/auth/"
+	return "https://api.snapcraft.io/"
 }
 
 var (
-	keyLength        = 4096
-	retryInterval    = 60 * time.Second
-	maxTentatives    = 15
-	deviceAPIBase    = deviceAPIBaseURL()
-	requestIDURL     = deviceAPIBase + "request-id"
-	serialRequestURL = deviceAPIBase + "devices"
+	keyLength     = 4096
+	retryInterval = 60 * time.Second
+	maxTentatives = 15
+	baseStoreURL  = baseURL()
 )
 
+func deviceAPIbase(proxyStore *asserts.Store, svcURL string) (s string) {
+	if proxyStore != nil && svcURL != "" {
+		// TODO: warning
+		logger.Noticef("UNSUPPORTED! trying to use a snap store proxy (at %q) together with a device service URL (%q). Falling back to contacting the service URL directly.", proxyStore.URL(), svcURL)
+	}
+	if svcURL != "" {
+		if svcURL[len(svcURL)-1] != '/' {
```

Comment:
Total nitpick: I find `strings.HasSuffix(..)` easier to read in such cases, but that's surely a personal preference.